fix(nuxt-img): access `prerender.env` only in server side

chadsr opened the pull request:
resolves #660

brunodeangelis commented:
Tested and indeed fixes the issue. Thank you @chadsr

slavanossar commented:
Could also just do

```js
if (process?.env?.prerender) {
  //...
}
```

chadsr commented:
> Could also just do
> 
> ```js
> if (process?.env?.prerender) {
>   //...
> }
> ```

Good point... I've switched it over to that syntax.
For some reason I was thinking optional chaining would still throw on `process` being undeclared, but it does not.

Flowko commented:
how come this not merged yet ?

slavanossar commented:
> > Could also just do
> > ```js
> > if (process?.env?.prerender) {
> >   //...
> > }
> > ```
> 
> Good point... I've switched it over to that syntax. For some reason I was thinking optional chaining would still throw on `process` being undeclared, but it does not.

Actually sorry, you are right about this. I just ran `nuxt generate` and when loading the page in the browser an error is thrown `ReferenceError: process is not defined`, wasn't happening when checking with `nuxt dev` which threw me off.

chadsr commented:
> > > Could also just do
> > > ```js
> > > if (process?.env?.prerender) {
> > >   //...
> > > }
> > > ```
> > 
> > 
> > Good point... I've switched it over to that syntax. For some reason I was thinking optional chaining would still throw on `process` being undeclared, but it does not.
> 
> Actually sorry, you are right about this. I just ran `nuxi generate` and when loading the page in the browser an error is thrown `ReferenceError: process is not defined`, wasn't happening when checking with `nuxi dev` which threw me off.

Looks like I feel for the same trap and only ran `nuxi dev` when checking as well.
I've reverted it back to checking with `typeof` now.

pi0 commented on src/runtime/components/nuxt-img.ts:
```diff
@@ -86,7 +86,7 @@ export default defineComponent({
     }
 
-    if (process.env.prerender) {
+    if (typeof process !== 'undefined' && process.env && process.env.prerender) {
```
```suggestion
    if (process.server && process.env.prerender) {
```

pi0 reviewed (approved):
Thanks!
